### PR TITLE
add docs for karmada components-rbac

### PR DESCRIPTION
**Pull request by B1F030**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
add docs for karmada components-rbac

**Which issue(s) this PR fixes**:
Part of [#5182](https://github.com/karmada-io/karmada/issues/5182)

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/reference/components-privileges.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: Components Privileges
```
Comment (zhzhuang-zju) [inline docs/reference/components-privileges.md:2]:
```suggestion
title: RBAC Permissions for Karmada Components
```

File: docs/reference/components-privileges.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+---
+
+### karmada-operator
+```yaml
+rules:
+  - apiGroups:
+      - coordination.k8s.io
+    resources:
+      - leases # karmada-operator requires access to the Lease resource for leader election.
+    verbs:
```
Comment (zhzhuang-zju) [inline docs/reference/components-privileges.md:12]: RBAC permission configuration for the component can be displayed in a table.

**Review by zhzhuang-zju — COMMENTED** [review]
@B1F030 This file would be more appropriately placed in the administrator/security directory.

File: docs/reference/components-privileges.md (hunk 1 above)
Comment (B1F030) [reply docs/reference/components-privileges.md]: Done.

File: docs/reference/components-privileges.md (hunk 2 above)
Comment (B1F030) [reply docs/reference/components-privileges.md]: Updated.

File: docs/administrator/security/rbac-permissions-for-karmada-components.md
Comment (RainbowMango) [inline docs/administrator/security/rbac-permissions-for-karmada-components.md:1]: How about using a shorter name?
Like `component-permission.md`

File: docs/administrator/security/rbac-permissions-for-karmada-components.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+---
+title: RBAC Permissions for Karmada Components
```
Comment (RainbowMango) [inline docs/administrator/security/rbac-permissions-for-karmada-components.md:2]:
```suggestion
title: Component Permission
```

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**